lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
I'm not sure what that means, though - if I pass a tuple `{:a, 1}` what are the allowed return values? Dialyzer generally treats values as separate types, until there are too many possibilities when it tries to collapse them down to a generic thing like "atom" or "integer".

What does it mean to be "of the same shape and contain same types on the same positions" in that case?